File: utils/jsonLoader.py
import os
import json
import logging
from functools import lru_cache

logger = logging.getLogger(__name__)

@lru_cache(maxsize=32)
def load_json_data(filename):
    """Charge un fichier JSON depuis le dossier data/ avec mise en cache ; retourne {} si erreur."""
    path = os.path.join('data', filename)
    try:
        with open(path, 'r', encoding='utf-8') as f:
            return json.load(f)
    except FileNotFoundError:
        logger.warning(f"⚠️ Fichier non trouvé: {path}")
        return {}
    except json.JSONDecodeError as e:
        logger.warning(f"⚠️ JSON invalide dans {path}: {e}")
        return {}
    except Exception as e:
        logger.exception(f"Erreur lecture {path}: {e}")
        return {}

File: utils/jsonLoader.js
import { readFileSync } from 'fs';
import { join, dirname } from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

/**
 * Load JSON data from the Data directory
 * @param {string} filename - Name of the JSON file
 * @returns {Object} Parsed JSON data
 */
export function loadJsonData(filename) {
    try {
        const filePath = join(__dirname, '..', 'Data', filename);
        const data = readFileSync(filePath, 'utf8');
        return JSON.parse(data);
    } catch (error) {
        console.error(`Error loading JSON file ${filename}:`, error);
        return {};
    }
}